Source/Dafny/Compiler-go.cs

```diff
@@ -1831,9 +1831,27 @@ private class SpecialNativeType : UserDefinedType {
     }
 
     protected override void EmitPrintStmt(TargetWriter wr, Expression arg) {
-      wr.Write("_dafny.Print(");
-      TrExpr(arg, wr, false);
-      wr.WriteLine(")");
+      bool isString = arg.Type.AsCollectionType != null &&
+                      arg.Type.AsCollectionType.AsSeqType != null &&
+                      arg.Type.AsCollectionType.AsSeqType.Arg.IsCharType;
+      if (isString) {
+        // needed for empty sequences known to be strings
+        if (arg.Resolved is MemberSelectExpr &&
+            (arg.Resolved as MemberSelectExpr).Member.IsExtern(out _, out _)) {
+          wr.Write("_dafny.Print((");
```

Comment:
There's a superfluous pair of parentheses on L1841 and L1843.
Also, this case is the same as L1851-1853, so maybe the nested `if` can be rewritten with 2 cases instead of 3.

Reply:
Done - though the complexity of the if condition does not lend itself to understanding.